### ISSUE TITLE
[HOLD for payment 2024-09-26][$250] Violations - "Receipt required" error message is missing from the expense details

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#47105]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.18-3
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team

**Issue was found when executing PR** https://github.com/Expensify/App/issues/44995#issue-2396295781
## Action Performed:
1. ND: Navigate to https://staging.new.expensify.com/
2. ND: Log in and validate a new Gmail account
3. ND: Create a workspace
4. ND: Switch to OD by navigating to Account settings - Cards & Domains
5. OD: Navigate to Settings - Workspaces - Group - Created workspace - Expenses
6. OD: Enable "Expense Violations"
7. OD: Upgrade the workspace to Control by clicking on the padlock icon
8. OD: Set "Receipt Required Amount" to any value but remain under the value set in "Max Expense Amount"
9. ND: Navigate to the workspace chat
10. ND: Navigate to Composers "+" button - Submit expense - Manual
11. ND: Set any value that's above "Receipt Required Amount" set in OD but under "Max Expense Amount"

## Expected Result:
"Receipt required" error message should be visible

## Actual Result:
"Receipt required" error message is missing from the expense details. The RBR shows in the LHN but the message is missing. Affects "Receipt Required Amount" amount set in OD Expenses tab and "Max Amount" set in a Category too

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/user-attachments/assets/10921411-5007-4089-88e4-d8d3ba647c35

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01da9c71405c9b9ab9</li>
        <li>Upwork Job ID: 1823792599432282200</li>
        <li>Last Price Increase: 2024-08-14</li>
<li>Automatic offers: </li>
<ul>
<li>akinwale | Reviewer | 103539096</li>
<li>ShridharGoel | Contributor | 103539097</li>
<li>dominictb | Contributor | 103689285</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @akinwale</details>

**Comment by cead22** [comment]
@ShridharGoel this case should've been covered in the tests for this PR https://github.com/Expensify/App/pull/45760, right? If so, can you take this over and fix it please?

**Comment by puneetlath** [comment]
@akinwale @dominictb @ShridharGoel maybe y'all can figure that out between you?

**Comment by puneetlath** [comment]
@akinwale any thoughts?

**Comment by puneetlath** [comment]
@akinwale bump.

**Comment by akinwale** [comment]
Missed the initial notification on this.

It looks like @dominictb's assessment is accurate. The proposed root cause and fix is not related to violations being in beta, so we can move forward with the [proposal](https://github.com/Expensify/App/issues/47105#issuecomment-2276892351) here.

🎀👀🎀 C+ reviewed.

**Comment by cead22** [comment]
> @puneetlath I don't think this issue is related to [PR](https://github.com/Expensify/App/pull/45760), because that PR is just to remove violations beta from front-end. Can you help check again?

@dominictb do the test steps in that PR not cover this scenario? If so, can you share how we need to update those test steps to make sure this case is covered? 

If the tests do cover this scenario, then I think it makes sense to consider this as a regression of that PR, since the reason this issue https://github.com/Expensify/App/issues/44995 was created with a higher bounty than the default, was to ensure the tests all passed.

That said, I don't feel super strongly about treating this as a regression vs considering it a new issue. I just wanted to provide the context and make sure that we update the test steps as part of the proposal

**Comment by puneetlath** [comment]
Looks like the [PR was deployed to production](https://github.com/Expensify/App/pull/48141#issuecomment-2362374474) 3 days ago. Is that right?

**Comment by akinwale** [comment]
> Looks like the [PR was deployed to production](https://github.com/Expensify/App/pull/48141#issuecomment-2362374474) 3 days ago. Is that right?

Yes, that seems to be the case which would make the payment due date 2024-09-26. The update must've been missed by automation.

**Comment by mallenexpensify** [comment]
Contributor: @dominictb paid $250 via Upwork
Contributor+: @akinwale paid $250 via Upwork

@akinwale plz complete the BZ checklist. 

**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@] Determine if we should create a regression test for this bug.
- [ ] [@] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@BZ] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by mallenexpensify** [comment]
@akinwale plz complete the BZ checklist above

**Comment by akinwale** [comment]
> * [x]  [@akinwale] The PR that introduced the bug has been identified. Link to the PR:
> * [x]  [@akinwale] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
> * [x]  [@akinwale] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

Not a regression.

> * [x]  [@akinwale] Determine if we should create a regression test for this bug.
> * [x]  [@akinwale] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.

**Regression Test Steps**
1. Launch Expensify
2. Log in with a new Gmail account
3. Create a Workspace
4. Navigate to Old Dot by accessing Account settings > Cards & Domains

_Old Dot_
5. Navigate to Settings > Workspaces > Group > "Created workspace" > Expenses
6. Enable Expense Violations
7. Upgrade the Workspace to a Control workspace
8. Set "Receipt Required Amount" to any value under the value set in "Max Expense Amount"
9. Switch back to New Dot

_New Dot_
10. Navigate to the workspace chat
11. Submit a new manual expense with an amount that is more than "Receipt Required Amount" and less than "Max Expense Amount"
12. Verify that the Receipt Required error message is displayed for the expense. 

Do we agree 👍 or 👎?

**Comment by akinwale** [comment]
@mallenexpensify Done!

**Comment by mallenexpensify** [comment]
Thx @akinwale 
- https://github.com/Expensify/Expensify/issues/435613
